TTHAnalysis/python/tools/eventVars_2lss.py

```diff
@@ -83,22 +82,28 @@ def run(self,event,Collection,metName):
                         ndr   += 1
                         sumdr += deltaR(j,j2)
                 ret["avg_dr_jet"] = sumdr/ndr if ndr else 0;
+
+            
+
+            met = getattr(event,metName+"_pt"+self.systsJEC[_var])
+            metphi = getattr(event,metName+"_phi"+self.systsJEC[_var])
+
             if nlep > 0:
                 ret["MT_met_lep1"] = sqrt( 2*leps[0].conePt*met*(1-cos(leps[0].phi-metphi)) )
             if nlep > 1:
                 ret["MT_met_lep2"] = sqrt( 2*leps[1].conePt*met*(1-cos(leps[1].phi-metphi)) )
-#                px = leps[0].conePt*cos(leps[0].phi) + leps[1].conePt*cos(leps[1].phi) + met*cos(metphi) 
-#                py = leps[0].conePt*sin(leps[0].phi) + leps[1].conePt*sin(leps[1].phi) + met*sin(metphi) 
-#                ht = leps[0].conePt + leps[1].conePt + met
-#                ret["MT_met_leplep"] = sqrt(max(0,ht**2 - px**2 - py**2))
-#            if nlep >= 1:
-#                sumapz, sumspz = 0,0
-#                for o in leps[:2] + jets:
-#                    pz = o.conePt*sinh(o.eta) if o in leps else o.pt*sinh(o.eta)
-#                    sumspz += pz
-#                    sumapz += abs(pz); 
-#                ret["sum_abspz"] = sumapz
-#                ret["sum_sgnpz"] = sumspz
+
+            if not _var and hasattr(event, '%s_pt_unclustEnUp'%metName):
+                met_up = getattr(event,metName+"_pt_unclustEnUp")
+                metphi_up = getattr(event,metName+"_phi_unclustEnUp")
+                met_down = getattr(event,metName+"_pt_unclustEnDown")
+                metphi_down = getattr(event,metName+"_phi_unclustEnDown")
+                if nlep > 0:
+                    allret["MT_met_lep1" + self.label + '_unclustEnUp'] = sqrt( 2*leps[0].conePt*met_up*(1-cos(leps[0].phi-metphi_up)) )
+                    allret["MT_met_lep1" + self.label + '_unclustEnDown'] = sqrt( 2*leps[0].conePt*met_down*(1-cos(leps[0].phi-metphi_down)) )
+                if nlep > 1:
+                    allret["MT_met_lep1" + self.label + '_unclustEnUp'] = sqrt( 2*leps[1].conePt*met_up*(1-cos(leps[1].phi-metphi_up)) )
+                    allret["MT_met_lep1" + self.label + '_unclustEnDown'] = sqrt( 2*leps[1].conePt*met_down*(1-cos(leps[1].phi-metphi_down)) )
```

Comment:
typo: lep2, not lep1 (unless I don't understand the code)

the MT_met_leplep and sum_abspx, sum_sgnpz have been intentionally removed as obsolete, or it's a mistake?

Reply:
yes, it was a typo. regarding MT_met_lep_lep etc, they were not used anymore (I just removed commented lines)